cognite_toolkit/_builtin_modules/bootcamp/ice_cream_api/functions/icapi_datapoints_extractor/handler.py

```diff
@@ -157,9 +157,9 @@ def handle(client: CogniteClient = None, data=None):
                     client.time_series.data.insert_multiple(datapoints=to_insert)
                     to_insert = []
 
-                report_ext_pipe(client, "success")
-            except Exception as e:
-                report_ext_pipe(client, "fail", e)
+        report_ext_pipe(client, "success")
+    except Exception as e:
+        report_ext_pipe(client, "fail", e)
 
         if to_insert:
```

Comment:
You may want to include this push in the try/catch. Beyond that this looks like a reasonable fix, thanks.